R/msf_dict_mortality.R

```diff
@@ -0,0 +1,76 @@
+# function to load MSF data dictionary for mortality surveys
+
+#' @importFrom rio import
+#' @importFrom epitrix clean_labels
+#' @importFrom tibble as_tibble
+#' @importFrom tidyr fill spread
+#' @importFrom dplyr mutate group_by row_number ungroup select recode
+#' @importFrom tidyselect everything
+#' @export
+#' @rdname msf_dict
+msf_dict_mortality <- function(name = "MSF-mortality_survey-dict.xlsx",
+                               tibble = TRUE) {
+
+  # get excel file path (need to specify the file name)
+  path <- system.file("extdata", name, package = "sitrep")
+
+  # read in data set - pasting the disease name for sheet
+  dat_dict <- rio::import(path, which = "Data Dictionary")
+
+  # clean col names
+  colnames(dat_dict) <- epitrix::clean_labels(colnames(dat_dict))
+
+  # fill NA values with previous non-NA value, replace "." in codes and names
+  dat_dict <- tidyr::fill(dat_dict,
+                          tidyselect::everything(), .direction = "down")
+
+  # minor tidying, e.g.: create "CodeX" assignments
+  dat_dict <- dplyr::mutate(dat_dict,
+    choice_code = replace(choice_code, choice_code == ".", NA),
+    choice_name = replace(choice_name, choice_name == ".", NA))
+  dat_dict <- dplyr::mutate(dat_dict,
+                            data_element_name = column_name,
+                            data_element_shortname = column_name,
```

Comment:
Come to think of it, I'm not sure what's happening here.... so what's happening here?

Reply:
I tried to make the `msf_dict_mortality` output similar to the `msf_dict` one. However, after your comment I checked again, and `data_element_name` and `data_element_shortname` are not needed, as we can define the  `varnames` in `gen_data` anyway. Will remove those duplications.